homeassistant/components/amcrest/camera.py

```diff
@@ -170,18 +208,183 @@ def turn_on(self):
         """Turn on camera."""
         self._enable_video_stream(True)
 
-    # Utility methods
+    def enable_motion_detection(self):
+        """Enable motion detection in the camera."""
+        self._enable_motion_detection(True)
+
+    def disable_motion_detection(self):
+        """Disable motion detection in camera."""
+        self._enable_motion_detection(False)
+
+    # Additional Amcrest Camera service methods
+
+    @callback
+    def async_enable_recording(self):
```

Comment:
Why are these callbacks? They should be coroutines since we want to await the task from the added job.

Reply:
For this and the previous comment I was just mirroring what was being done in `homeassistant/components/camera/__init__.py` assuming that was correct.

Would it be correct this way?
```python
    async def async_enable_recording(self):
        """Call the job and enable recording."""
        return self.hass.async_add_executor_job(self._enable_recording, True)
```
Or does it need to be? (Sorry, still not 100% understanding asyncio.)
```python
    async def async_enable_recording(self):
        """Call the job and enable recording."""
        return await self.hass.async_add_executor_job(self._enable_recording, True)
```

Reply:
the latter. Camera doing a callback with return async_add_job was a "shortcut" that messes with stacktraces and we're trying to undo.

Reply:
Actually I think it should be:
```python
    async def async_enable_recording(self):
        """Call the job and enable recording."""
        await self.hass.async_add_executor_job(self._enable_recording, True)
```
because the "normal" (i.e., non-coroutine) function ( `self._enable_recording` in this case) doesn't actually return anything. The return was only there because the callback was originally returning the task to be waited on. But now that is no longer the case, right?

The bigger picture is (and I'll use one of the other examples that is more straightforward) this coroutine (in `amcrest/__init__.py`):
```python
    async def async_goto_preset(service):
        preset = service.data[ATTR_PRESET]
        for camera in await async_extract_from_service(service):
            await camera.async_goto_preset(preset)
```
is calling (in `amcrest/camera.py`):
```python
    async def async_goto_preset(self, preset):
        """Call the job and move camera to preset position."""
        await self.hass.async_add_executor_job(self._goto_preset, preset)
```
So you can see there's nothing to return. Also, now I'm if there should even _be_ an `await` in the second function??? I would guess not and it really should be this:
```python
    async def async_goto_preset(self, preset):
        """Call the job and move camera to preset position."""
        self.hass.async_add_executor_job(self._goto_preset, preset)
```